File: app/src/main/java/com/example/mobv2/ui/callbacks/GetCommentCallback.java
package com.example.mobv2.ui.callbacks;

import android.util.Log;

import com.example.mobv2.adapters.CommentsAdapter;
import com.example.mobv2.models.Comment;
import com.example.mobv2.serverapi.MOBServerAPI;
import com.google.gson.internal.LinkedTreeMap;

public class GetCommentCallback implements MOBServerAPI.MOBAPICallback
{
    private final CommentsAdapter commentsAdapter;

    public GetCommentCallback(CommentsAdapter commentsAdapter)
    {
        this.commentsAdapter = commentsAdapter;
    }

    @Override
    public void funcOk(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());

        var response = (LinkedTreeMap<String, Object>) obj.get("response");

        Comment comment = Comment.parseFromMap(response);
        commentsAdapter.addComment(comment);
    }

    @Override
    public void funcBad(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
    }

    @Override
    public void fail(Throwable obj)
    {
        Log.v("DEBUG", obj.toString());
    }
}

File: app/src/main/java/com/example/mobv2/callback/GetCommentCallback.kt
package com.example.mobv2.callback

import android.util.Log
import com.example.mobv2.callback.abstraction.GetCommentFailCallback
import com.example.mobv2.callback.abstraction.GetCommentOkCallback
import com.example.mobv2.ui.activity.MainActivity
import com.example.mobv2.util.abstraction.responseStateCallback.HasFailStateCallback
import com.example.mobv2.util.abstraction.responseStateCallback.HasOkStateCallback
import com.google.gson.internal.LinkedTreeMap
import serverAPI.MOBServerAPI.MOBAPICallback

class GetCommentCallback(private val mainActivity: MainActivity) : MOBAPICallback,
    HasOkStateCallback<GetCommentOkCallback>, HasFailStateCallback<GetCommentFailCallback>
{
    private var okCallback: GetCommentOkCallback? = null
    private var failCallback: GetCommentFailCallback? = null
    override fun funcOk(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
        val response = obj["response"] as LinkedTreeMap<String, Any>

        okCallback?.parseCommentFromMapAndAddToComments(response)
    }

    override fun funcBad(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun fail(obj: Throwable)
    {
        Log.v("DEBUG", obj.toString())

        failCallback?.onDisconnect()
    }

    override fun setOkCallback(okCallback: GetCommentOkCallback)
    {
        this.okCallback = okCallback
    }

    override fun setFailCallback(failCallback: GetCommentFailCallback)
    {
        this.failCallback = failCallback
    }
}